sia/src/rhp/rpc.rs

```diff
@@ -0,0 +1,613 @@
+        self.tip_height.encode(&mut state).unwrap();
+        self.valid_until.encode(&mut state).unwrap();
+        state.finalize().into()
+    }
+
+    /// Checks if the prices are valid for the given host key and timestamp.
+    pub fn is_valid(&self, host_key: &PublicKey, timestamp: OffsetDateTime) -> bool {
+        self.valid_until > timestamp
+            && self.tip_height > 0
+            && host_key.verify(&self.sig_hash(), &self.signature)
+    }
+}
+
+/// Contains the settings of a host, including its prices and other parameters.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct HostSettings {
+    /// The version of the protocol the host is using.
+    pub protocol_version: [u8; 3],
+    /// The current release the host is running.
+    pub release: String,
+    /// The wallet address of the host to use for contract payments.
+    pub wallet_address: Address,
+    /// If the host is accepting new contracts.
+    pub accepting_contracts: bool,
+    /// The maximum amount of collateral that the host will accept for a
+    /// single contract.
+    pub max_collateral: Currency,
+    /// The maximum duration, in blocks, that the host will accept for a contract.
+    pub max_contract_duration: u64,
+    /// The amount of storage, in sectors, that the host has available.
+    pub remaining_storage: u64,
+    /// The total amount of storage, in sectors, that the host is offering.
+    pub total_storage: u64,
+    /// The current prices of the host
+    pub prices: HostPrices,
+}
+
+/// An account token is used to pay for RPC calls that do not
+/// require a contract.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct AccountToken {
+    pub host_key: PublicKey,
+    pub account: PublicKey,
+    pub valid_until: OffsetDateTime,
+
+    pub signature: Signature,
+}
+
+impl AccountToken {
+    fn compute_sig_hash(
+        host_key: &PublicKey,
+        account: &PublicKey,
+        valid_until: &OffsetDateTime,
+    ) -> Hash256 {
+        let mut state = Params::new().hash_length(32).to_state();
+        host_key.encode(&mut state).unwrap();
+        account.encode(&mut state).unwrap();
+        valid_until.encode(&mut state).unwrap();
+        state.finalize().into()
+    }
+
+    pub fn new(account_key: &PrivateKey, host_key: PublicKey) -> Self {
+        let expiration_time = OffsetDateTime::now_utc() + time::Duration::minutes(5);
+        let sig_hash =
+            Self::compute_sig_hash(&host_key, &account_key.public_key(), &expiration_time);
+        AccountToken {
+            host_key,
+            account: account_key.public_key(),
+            valid_until: expiration_time,
+
+            signature: account_key.sign_hash(&sig_hash),
+        }
+    }
+
+    pub fn sig_hash(&self) -> Hash256 {
+        Self::compute_sig_hash(&self.host_key, &self.account, &self.valid_until)
+    }
+}
```

Comment:
Maybe an associated function to generate an account makes sense here? Unless we have that somewhere else already.

Reply:
An account in our Rust SDK is just a public key. It doesn't have a concrete type like in Go.